test/unit/BondingManager.js

```diff
@@ -322,6 +313,80 @@ contract("BondingManager", accounts => {
             await expectThrow(bondingManager.bond(1000, transcoder0, {from: delegator}))
         })
 
+        describe("update transcoder pool", () => {
+            beforeEach(async () => {
+                await bondingManager.bond(500, transcoder2, {from: transcoder2})
+                await bondingManager.transcoder(5, 10, {from: transcoder2})
+                await fixture.roundsManager.setMockUint256(functionSig("currentRound()"), currentRound+1)
+                await fixture.roundsManager.execute(bondingManager.address, functionSig("setCurrentRoundTotalActiveStake()"))
+            })
+
+            it("adds a new transcoder to the pool", async () => {
+                await bondingManager.bond(1000, transcoder2, {from: delegator})
+                const firstTranscoder = await bondingManager.getFirstTranscoderInPool()
+                const firstStake = await bondingManager.transcoderTotalStake(firstTranscoder)
+                const secondTranscoder = await bondingManager.getNextTranscoderInPool(firstTranscoder)
+                const secondStake = await bondingManager.transcoderTotalStake(secondTranscoder)
+                const firstDel = await bondingManager.getDelegator(firstTranscoder)
+                const secondDel = await bondingManager.getDelegator(secondTranscoder)
+                assert.equal(firstTranscoder, transcoder1)
+                assert.equal(firstStake.toString(), firstDel.delegatedAmount.toString())
+                assert.equal(secondTranscoder, transcoder2)
+                assert.equal(secondStake.toString(), secondDel.delegatedAmount.toString())
+            })
+
+            describe("evicts a transcoder from the pool", () => {
+                it("last transcoder gets evicted and new transcoder gets inserted", async () => {
+                    bondingManager.TranscoderEvicted({}).on("data", e => {
+                        assert.equal(e.returnValues.transcoder, transcoder0)
```

Comment:
Can we confirm that this test is actually being run (or is the test completing before the event emitter fires)?

Reply:
Test is completing before the event emitter fires, want me to address this in this PR or in #326 ?

Reply:
fixed in 7fddb5f